### PR TITLE
feat: add an optional trace header to metrics/logging

### LINKED ISSUE #145: Add trace context to metric and logging calls
Per Ops, we should add the trace header data to the metric and logging calls.

This header may be `X-Cloud-Trace-Context`, ops to confirm.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/CONSVC-180)

### PR DESCRIPTION
**Pull request by jrconlin**
## Description

This adds a `trace_header` setting. This will add the value (if found in
the Headers) to the tags as `header.trace`.

(e.g. if a request has `X-Cloud-Trace-Context: abc123` this will add a
tag with `header.trace: abc123` to the outboud tags for logging and
metrics.)
## Testing

Ensure the content of the tracing header is sent to metric and logging.

## Issue(s)

Closes #145

### CODE REVIEW

File: src/settings.rs [hunk 1]
```diff
@@ -204,6 +208,23 @@ impl Settings {
     }
 }
 
+// TODO: someone better at lifetimes could fix these.
+// The returned settings need only last as long as the request.
+// we clone to break out of the Arc
+impl From<&HttpRequest> for Settings {
```
Comment (pjenvey) [inline src/settings.rs:214]: This was trickier than I thought it was gonna be 😅 but I think this works:

https://github.com/mozilla-services/contile/compare/feat/145-trace...feat/145-trace-lifetimes

Wanna apply the same to `impl From<&ServiceRequest> for Settings` below?

Comment (jrconlin) [reply src/settings.rs:214]: Oh, cool! Thanks!
Yep, I'll modify the other ones along the same lines.

**Comment by jbuck** [comment]
@jrconlin sorry, I don't understand this code so I'm not sure where to add my comment... but it's critical that the trace tag _isn't_ added to statsd, and only added to sentry/log output. The trace id is a random string which will blow up influxdb cardinality

**Comment by jrconlin** [comment]
@jbuck yeah, that's exactly the sort of comment I need. thanks!

File: src/tags.rs [hunk 2]
```diff
@@ -127,6 +129,15 @@ impl From<&RequestHead> for Tags {
                 extra.insert("ua".to_owned(), uas.to_string());
             }
         }
+        if let Some(tracer) = settings.trace_header.clone() {
```
Comment (pjenvey) [inline src/tags.rs:132]: nit

```suggestion
        if let Some(ref tracer) = settings.trace_header {
```

**Review by pjenvey — APPROVED** [review]

**Review by pjenvey — APPROVED** [review]

**Pull request merged**